luarules/gadgets/game_tax_resource_sharing.lua

```diff
@@ -56,27 +69,64 @@ function gadget:AllowResourceTransfer(senderTeamId, receiverTeamId, resourceType
+	local actualSentAmount = 0  -- How much sender actually loses
+	local actualReceivedAmount = 0 -- How much receiver actually gains
+	local currentCumulative = 0 -- Declare here to ensure scope
+
+	-- Apply cumulative threshold logic only for metal
+	if resourceName == 'metal' and metalTaxThreshold > 0 then
+		currentCumulative = cumulativeMetalSent[senderTeamId] or 0 -- Assign value here
+
+		local allowanceRemaining = math.max(0, metalTaxThreshold - currentCumulative)
+		local untaxedPortion = math.min(transferAmount, allowanceRemaining)
+		local taxablePortion = transferAmount - untaxedPortion
+
+		if taxablePortion > 0 then
+			-- Apply tax only to the taxable portion
+			local taxedPortionReceived = taxablePortion * (1 - sharingTax)
+			local taxedPortionSent = taxablePortion / (1 - sharingTax) -- Assuming tax isn't 100%
+			if sharingTax == 1 then taxedPortionSent = taxablePortion end -- Handle 100% tax case
+
+			actualReceivedAmount = untaxedPortion + taxedPortionReceived
+			actualSentAmount = untaxedPortion + taxedPortionSent
+		else
+			-- Entire transfer is within the remaining allowance, no tax
+			actualReceivedAmount = untaxedPortion -- which equals transferAmount
+			actualSentAmount = untaxedPortion     -- which equals transferAmount
+			currentSharingTax = 0 -- Log purpose
+		end
+
+	else -- Energy transfer OR Metal Threshold is 0
+		actualReceivedAmount = transferAmount * (1 - currentSharingTax)
+		actualSentAmount = actualReceivedAmount / (1 - currentSharingTax) -- Assuming tax isn't 100%
+		if currentSharingTax == 1 then actualSentAmount = transferAmount end -- Handle 100% tax case
+	end
 
-	Spring.SetTeamResource(receiverTeamId, resourceName, rCur+taxedAmount)
-	local sCur, _, _, _, _, _ = Spring.GetTeamResources(senderTeamId, resourceName)
-	Spring.SetTeamResource(senderTeamId, resourceName, sCur-totalAmount)
+	-- Ensure we don't send more than originally intended due to tax calculation edge cases / maxShare limit
+	actualSentAmount = math.min(actualSentAmount, amount) 
+	actualReceivedAmount = math.min(actualReceivedAmount, transferAmount)
 
-	-- Block the original transfer
-	return false
-end
+	-- Perform the transfer
+	Spring.SetTeamResource(receiverTeamId, resourceName, rCur + actualReceivedAmount)
+	local sCur, _, _, _, _, _ = Spring.GetTeamResources(senderTeamId, resourceName)
+	Spring.SetTeamResource(senderTeamId, resourceName, sCur - actualSentAmount)
 
-function gadget:AllowUnitTransfer(unitID, unitDefID, oldTeam, newTeam, capture)
-	local unitCount = spGetTeamUnitCount(newTeam)
-	if capture or spIsCheatingEnabled() or unitCount < gameMaxUnits then
-		return true
+	-- Update cumulative total *after* successful transfer (only for metal)
+	if resourceName == 'metal' and metalTaxThreshold > 0 then
+		local updatedCumulative = currentCumulative + actualSentAmount -- Track amount SENT by this sender
+		cumulativeMetalSent[senderTeamId] = updatedCumulative -- Update sender's total
 	end
+
 	return false
 end
 
-
```

Comment:
What do you guys think about moving the block guarding allied bots over to the `game_disable_assist_ally` and we keep resource tax PURELY focused on resource taxes. These hiding here is non-obvious behavior IMO

Reply:
Eh I think I changed my mind here. This PR seems scoped correctly.